core/test/matrix/fbcsr_sample.hpp

```diff
@@ -0,0 +1,722 @@
+    using MatData = gko::matrix_data<value_type, index_type>;
+    using SparCsr = gko::matrix::SparsityCsr<value_type, index_type>;
+
+
+    const size_type nrows = 6;
+    const size_type ncols = 12;
+    const size_type nnz = 36;
+    const size_type nbrows = 2;
+    const size_type nbcols = 4;
+    const size_type nbnz = 4;
+    const int bs = 3;
+    const std::shared_ptr<const gko::Executor> exec;
+
+
+    FbcsrSample(std::shared_ptr<const gko::ReferenceExecutor> rexec)
+        : exec(rexec)
+    {}
+
+    /**
+     * @return The sample matrix in FBCSR format
+     */
+    std::unique_ptr<Fbcsr> generate_fbcsr() const
+    {
+        std::unique_ptr<Fbcsr> mtx =
+            Fbcsr::create(exec,
+                          gko::dim<2>{static_cast<size_type>(nrows),
+                                      static_cast<size_type>(ncols)},
+                          nnz, bs);
+
+        value_type *const v = mtx->get_values();
+        index_type *const c = mtx->get_col_idxs();
+        index_type *const r = mtx->get_row_ptrs();
+        r[0] = 0;
+        r[1] = 2;
+        r[2] = 4;
+        c[0] = 1;
+        c[1] = 3;
+        c[2] = 0;
+        c[3] = 2;
+
+        gko::blockutils::DenseBlocksView<value_type, index_type> vals(v, bs,
+                                                                      bs);
+
+        if (mtx->get_size()[0] % bs != 0)
+            throw gko::BadDimension(__FILE__, __LINE__, __func__, "test fbcsr",
+                                    mtx->get_size()[0], mtx->get_size()[1],
+                                    "block size does not divide the size!");
+
+        for (index_type ibrow = 0; ibrow < mtx->get_num_block_rows(); ibrow++) {
+            const index_type *const browptr = mtx->get_row_ptrs();
+            for (index_type inz = browptr[ibrow]; inz < browptr[ibrow + 1];
+                 inz++) {
+                const index_type bcolind = mtx->get_col_idxs()[inz];
+                const value_type base = (ibrow + 1) * (bcolind + 1);
+                for (int ival = 0; ival < bs; ival++)
+                    for (int jval = 0; jval < bs; jval++)
+                        vals(inz, ival, jval) =
+                            base + static_cast<gko::remove_complex<value_type>>(
+                                       ival * bs + jval);
+            }
+        }
+
+        // Some of the entries are set to zero
+        vals(0, 2, 0) = gko::zero<value_type>();
+        vals(0, 2, 2) = gko::zero<value_type>();
+        vals(3, 0, 0) = gko::zero<value_type>();
+
+        v[34] += FBCSR_TEST_IMAGINARY;
+        v[35] += FBCSR_TEST_IMAGINARY;
+
+        return mtx;
+    }
+
+    /**
+     * @return Sample matrix in CSR format
+     *
+     * Keeps explicit zeros.
+     */
+    std::unique_ptr<Csr> generate_csr() const
+    {
+        std::unique_ptr<Csr> csrm =
+            Csr::create(exec, gko::dim<2>{nrows, ncols}, nnz,
+                        std::make_shared<typename Csr::classical>());
```

Comment:
Is it possible to create an array for these and create a CSR matrix directly from the arrays. These one line value assignments make it unnecessarily long. I mean using [this constructor of CSR](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/matrix/csr.hpp#L897)